lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -551,13 +590,55 @@ def _log_notices(self, cursor):
 
 
 class PGCompiler_psycopg2(PGCompiler):
-    pass
+    def __init__(
+        self, dialect, statement, column_keys=None, inline=False, **kwargs
+    ):
+        self.dialect = dialect
+        self.multiple_rows = inline
+        self.execute_values_insert_template = None
```

Comment:
this can be set as a class-level variable to start wtih, as None, that would allow you to not have to rewrite the whole constructor:

```
class PGCompiler_psycopg2(PGCompiler):
    execute_values_insert_template = None

```

Reply:
OK. You mean I guess all those variables?

Reply:
But what about the other two variables? I still need to override the constructor to get those

Reply:
And also, is this class guaranteed to have only one instance? If not it's problematic to use a class variable to hold the template

Reply:
So there are the "other two variables", "dialect", and "multiple_rows", for which I don't see the purpose for these assignments.  The PGCompiler_psycopg2 object already has a .dialect instance variable you can access as "self.dialect".   The "inline" keyword argument is also available on the object as just that, "self.inline", there is no need to copy it to an additional instance variable.

For "execute_values_insert_template", it appears you only modify execute_values_insert_template in the generate_values_placeholders_str method.   You can safely default this to None at the class level.  When you later invoke "self.execute_values_insert_template = 'some value'", that sets an *instance level* variable which overrides the class-level default version.    SQLAlchemy uses this pattern all over the place:

```
Python 3.7.3 (default, May 11 2019, 00:38:04) 
[GCC 9.1.1 20190503 (Red Hat 9.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo:
...    some_arg = None
... 
>>> 
>>> f1 = Foo()
>>> f1.some_arg
>>> f2 = Foo()
>>> f2.some_arg = 'a'
>>> f3 = Foo()
>>> f3.some_arg = 'b'
>>> f1.some_arg
>>> f2.some_arg
'a'
>>> f3.some_arg
'b'

```


If I'm misreading or misinterpreting what I'm seeing here, that is very possible.    So let me know if I'm doing that here.

Reply:
OK. I missed the dialect variable, and didn't know about this behavior of class variables.

Reply:
As for multiple_rows, I wasn't sure that reading self.inline the correct way to get this info. But since you didn't comment about that, I guess it is :)